datafusion/functions-aggregate-common/benches/null_state_accumulate.rs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+extern crate criterion;
+
+use std::sync::Arc;
+
+use arrow::array::ArrowNativeTypeOp;
+use arrow::{
+    array::{ArrayRef, AsArray, BooleanArray, Int64Array},
+    datatypes::Int64Type,
+};
+use criterion::{criterion_group, criterion_main, Criterion};
+use datafusion_functions_aggregate_common::aggregate::groups_accumulator::group_index_operations::FlatGroupIndexOperations;
+use datafusion_functions_aggregate_common::aggregate::groups_accumulator::{
+    accumulate::{self, accumulate_indices, NullStateAdapter},
+    blocks::GeneralBlocks,
+    group_index_operations::{BlockedGroupIndexOperations, GroupIndexOperations},
+};
+
+fn generate_group_indices(len: usize) -> Vec<usize> {
+    (0..len).collect()
```

Comment:
Does using indices `0..len` mean this benchmark is  testing the case where each input has a new unique group?

If this is the case, then I think it would make sense that the benchmark shows lots of time allocating / zeroing memory: the bulk of the work will be copying each value into a new accumulator slot.